### PR TITLE
resizingDone

### PR DESCRIPTION
**Pull request by ArtemAAA**

### CODE REVIEW

File: src/js/core/views.js [hunk 1]
```diff
@@ -109,3 +110,94 @@ export class UIView {
...
+    this.resizeHoldFalse();
+    document.body.style.cursor = "default";
+  }
+
+  // check if mouse position x is close to menu border
+  checkCloseEnough(mouseX, elementLocationX) {
+    return Math.abs(mouseX - elementLocationX) < this.element.closeEnough;
+  }
+
+  hideMenuRight() {
```
Comment (alexander-vielimchanitsia) [inline src/js/core/views.js]: seems it doesn't work

and bad name

File: src/js/core/views.js [hunk 2]
```diff
@@ -109,3 +110,94 @@ export class UIView {
...
+    let item;
+    for (item in this.element.children){
+      if (constants.canvas.width - this.element.location.x <=
+          constants.canvasCtx.measureText(this.element.children[item].text).width){
```
Comment (alexander-vielimchanitsia) [inline src/js/core/views.js]: it can have not only text elements

File: src/js/core/views.js [hunk 3]
```diff
@@ -109,3 +110,94 @@ export class UIView {
...
+        let item;
+        for (item in this.element.children){
+          this.element.children[item].hide()
+        }
+        return true
+      }
+    } 
+  }
+
+  showMenuRight() {
```
Comment (alexander-vielimchanitsia) [inline src/js/core/views.js]: bad name

File: src/js/core/views.js [hunk 4]
```diff
@@ -109,3 +110,94 @@ export class UIView {
...
+    this.element.show()
+    let item;
+    for (item in this.element.children){
```
Comment (alexander-vielimchanitsia) [inline src/js/core/views.js]: `for (let item in this.element.children)`

File: src/js/core/views.js [hunk 5]
```diff
@@ -109,3 +110,94 @@ export class UIView {
...
+
+  // check if mouse position x is close to menu border
+  checkCloseEnough(mouseX, elementLocationX) {
+    return Math.abs(mouseX - elementLocationX) < this.element.closeEnough;
+  }
+
+  hideMenuRight() {
+    let item;
+    for (item in this.element.children){
+      if (constants.canvas.width - this.element.location.x <=
```
Comment (alexander-vielimchanitsia) [inline src/js/core/views.js]: this check shouldn't be there

File: src/js/core/views.js (hunk 1 above)
Comment (alexander-vielimchanitsia) [reply src/js/core/views.js]: correct `UIElement.hide` method instead

File: src/js/core/views.js (hunk 3 above)
Comment (alexander-vielimchanitsia) [reply src/js/core/views.js]: correct `UIElement.show` method instead

File: src/js/core/views.js [hunk 6]
```diff
@@ -109,3 +110,94 @@ export class UIView {
...
+      let resize = mousePosition.x;
+      let resizeWidth = this.element.width + this.element.location.x - resize;
+
+      this.element.location.x = resize;
+
+      this.element.setWidth(resizeWidth);
+
+      // clear and draw to have fluent resizing
+      constants.canvasCtx.clearRect(0, 0, constants.canvas.width, constants.canvas.height);
+      super.draw(this.element)
```
Comment (alexander-vielimchanitsia) [inline src/js/core/views.js]: it shouldn't be there, just need to change element's params and set` isValidCanvasState` to `false`

**Comment by ArtemAAA** [comment]
Nice!

**Pull request closed without merging**